sdk/core/Azure.Core/src/Http/HttpRequestOptions.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.Http
+{
+    /// <summary>
+    /// Specifies standard HTTP request options.
+    /// </summary>
+    public struct HttpRequestOptions : IEquatable<HttpRequestOptions>
```

Comment:
Why did we move away from the suffix `Conditions`?  Storage uses `Options` for things like "how much data do you want" and `Conditions` for placing restrictions on the operation.  I think it also matches better conceptually since these are always called preconditions.

Reply:
This was proposed by Johan.  The thinking was that if there are other options we wanted to add to the request (unrelated to precondition headers), this would be a place to add those in the future in a general way.  Are the Options in Storage specific to the HTTP request?  If not, they could be some type of storage-specific options.  StorageRequestOptions, e.g.?

Reply:
`Options` in Storage are specific to the operation _(do you want blob metadata when we give you back the blob info?)_ and we already have them in custom types.  We also have multiple `Conditions` classes because there are other constraints you can place on a request _(ensure the my lease ID is still locking down this resource)_.

I'd rather not have a catchall type for "other request stuff."  Was this @johanste wanting a generic "other headers bag"?  I'd prefer a type specific to HTTP access conditions.  Worst case, we could always add an instance of this in the catchall type if we ever add one and it would still be useful in scoping HTTP access conditions within the broader `RequestStuff` class.

Reply:
Chatted with Krzysztof on this ... RequestOptions is similar to ClientOptions in this proposal.  If you want to add specific functionality to it, you can subclass from it and add those things, or if functionality is universal to requests, we add it to RequestOptions and everything gets it.  It is a catchall type, but it means there isn't a lot of API churn around the need to add parameters to API calls as services add functionality around requests.

Reply:
As an FYI, @tg-msft, I'm looking at making the values in RequestOptions more consistent with the naming [Johan suggested](https://github.com/Azure/azure-sdk/issues/638#issuecomment-534264810) on the conditional request issue, which captures semantics more clearly than direct use of the HTTP header names.

Reply:
I think this makes the Storage APIs harder to use.  Today we only expose those headers on the operations that support them.  If we put them on a catchall type, folks will have to rely on documentation to figure out when they're valid.  _(And we can't use the friendly bool alternatives for most operations because we don't have a model with an ETag.)_

Reply:
@tg-msft , pushed new version called ConditionalRequestOptions.  LMK what you think.

Reply:
fyi @KrzysztofCwalina